// aznumconv/index.js
function getDigit(i,num){

   return  parseInt(parseInt((parseInt(num) / 10 ** (i-1))) % 10);
}

function areDigitsZero(num){
  var length = num.toString().length;

  for (var i = 1; i < length; i++) {
    if(num.toString().charAt(i) != 0){
      return false;
    }
  }
  return true;
}


exports.convertNormal  = function convertFromNumberToQuantitative(num){
  var tp = [" yüz"," min"," milyon"," milyard"," trilyon"," trilyard"];
  var  dec = ["", " bir", " iki", " üç", " dörd", " beş", " altı", " yeddi", " səkkiz", " doqquz"];
  var  ten = ["", " on", " iyirmi", " otuz", " qırx", " əlli", " altmış", " yetmiş", " səksən", " doxsan"];
  var text = "";
  var length = num.toString().length;
  if(num == 0 ){
    return "sıfır";

  }

  for(i=length; i >0; i--){
    var digit = getDigit(i,num);
    if(i % 3 == 0 ){
      if(digit == 0){
        text += "";
      }
      else if(digit == 1){
        text += tp[0];
      }
      else{
        text += dec[digit] + tp[0];

      }
    }
    else if(i % 3 == 1){
      if(i > 3 && !getDigit(i,num) && !getDigit(i + 1,num) && !getDigit(i + 2,num)){
        text += "";
      }
      else if(i > 3){
        if(digit == 1 && (areDigitsZero(num) || length == 4)) {
          text+= tp[Math.floor(i / 3)];
        }else{
        text += dec[digit] + tp[Math.floor(i / 3)];
      }

      }
      else{
        text += dec[digit];
      }
    }
    else if(i % 3 == 2){
      if(digit == 0){
        text += "";

      }
      else {
        text += ten[digit];
      }
    }
  }
  return text;

}










var SUFFIX = {"frontNonRounded":'inci', "frontRounded":'üncü', "backNonRounded":'ıncı',"backRounded" :'uncu'};


var frontNonRounded = ['i','e','ə'];
var frontRounded = ['ü','ö'];
var backNonRounded = ['a','ı'];
var backRounded = ['u','o'];
var vowels = frontNonRounded.concat(frontRounded,backRounded,backNonRounded);

var suffix = '';
exports.convertOrdinal = function convertToOrdinal(text){
  if(text == "sıfır"){
    return "sıfrıncı";
  }
  var lastWord = text.split(' ').splice(-1).join('');
  var vowel = '';
  var reversed = lastWord.split('').reverse().join('');
  for(i = 0;i < reversed.length; i++){
    if(vowels.includes(reversed.charAt(i))){
        vowel = reversed.charAt(i);
        break;
    }
  }

  if(frontNonRounded.includes(vowel)){
    suffix = SUFFIX.frontNonRounded;
  }
  else if(frontRounded.includes(vowel)){
    suffix = SUFFIX.frontRounded;
  }
  else if(backRounded.includes(vowel)){
    suffix = SUFFIX.backRounded;
  }
  else if(backNonRounded.includes(vowel)){
    suffix = SUFFIX.backNonRounded;
  }

  if(reversed.charAt(0) == suffix.charAt(0)){
    suffix = suffix.substr(1);
  }
  var ordinal = text += suffix;

  return ordinal;
}
